buildkite/resource_organization_settings.go

```diff
@@ -0,0 +1,101 @@
+package buildkite
+
+import (
+  "context"
+  "fmt"
+
+  "github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+  "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceOrganizationSettings() *schema.Resource {
+  return &schema.Resource{
+    CreateContext: CreateUpdateDeleteOrganizationSettings,
+    ReadContext:   ReadOrganizationSettings,
+    UpdateContext: CreateUpdateDeleteOrganizationSettings,
+    DeleteContext: CreateUpdateDeleteOrganizationSettings,
+    Importer: &schema.ResourceImporter{
+      State: schema.ImportStatePassthrough,
+    },
+
+    Schema: map[string]*schema.Schema{
+      "id": {
+        Type:     schema.TypeString,
+        Computed: true,
+      },
+      "slug": {
+        Type:     schema.TypeString,
+        Required: true,
+      },
+      "uuid": {
+        Type:     schema.TypeString,
+        Computed: true,
+      },
+      "name": {
+        Type:     schema.TypeString,
+        Computed: true,
+      },
+      "allowed_api_ip_addresses": {
+        Type:     schema.TypeString,
+        Optional: true,
+      },
+    },
+  }
+}
+
+func assertError(err error) diag.Diagnostics {
+  if err != nil {
+    return diag.FromErr(err)
+  }
+  return nil
+}
+
+// CreateUpdateDeleteOrganizationSettings is used for the creation, updating and deleting of a Buildkite organization's settings
+// In the future, this will be split into separate functions, but given it only has one field, it's not worth it yet
+func CreateUpdateDeleteOrganizationSettings(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+  var diags diag.Diagnostics
+  client := m.(*Client)
+
+  response, err := getOrganization(client.genqlient, d.Get("slug").(string))
+
+  assertError(err)
+
+  if response.Organization.Id == "" {
+    return diag.FromErr(fmt.Errorf("organization not found: '%s'", d.Get("slug")))
+  }
```

Comment:
I think you're doing this because the mutation below requires the org ID and not the slug (which we already have through configuration of the provider). We are already doing this in at least one other place so maybe its time to extract this somewhere else - I'll create another ticket to follow up on this though